corehq/apps/users/models.py

```diff
@@ -382,6 +382,7 @@ class DomainMembership(Membership):
     override_global_tz = BooleanProperty(default=False)
     role_id = StringProperty()
     location_id = StringProperty()
+    assigned_location_ids = StringListProperty()
```

Comment:
I have implemented it already, but haven't exposed the UI as there was no UI for location currently. It will be easier to add a UI latter when needed.